### DISCUSSION TITLE
[Question] Does Direct Workflow have an action curriculum feature?

### DISCUSSION THREAD

**Discussion by H-Hisamichi** [isaac-sim/IsaacLab#1478] [Q&A] [answered]
Hello everyone,

I'm training Hexapod based on ANYmal C direct RL code, Hexapod has more leg freedom than Quadropod.
However, it is not stable when operating at maximum angle from the beginning of training. So I try to introduce a curriculum to the output of the action. I found `omni.isaac.lab.envs.mdp.curriculums` but this seems to be a feature of manager based RL.
Is there a way to introduce curriculum into actions in direct RL?

Thank you!

**Comment by kellyguo11** [comment, marked as answer]
For direct workflow, you can implement the curriculum logic directly in the environment code. You are free to design the curriculum in any way that's suitable for the task, such as increasing complexity of the task, rebalancing reward weights, using different reward functions, etc...

**Reply by H-Hisamichi** [reply]
Hello @kellyguo11,

Thank you for any advice!
Do you know any sample code where a curriculum is applied directly workflow?

**Reply by phsilvarepo** [reply]
Hello @H-Hisamichi,
Did you manage to implement this, and if so could you show an example?

**Reply by H-Hisamichi** [reply]
Hi @phsilvarepo,

I have ultimately switched everything to a `manager-based workflow`.
